datalad/support/sshconnector.py

```diff
@@ -111,15 +112,16 @@ def open(self):
         connection, if it is not there already.
         """
 
-        # set control options
-        ctrl_options = ["-o", "ControlMaster=auto", "-o", "ControlPersist=yes"] + self.ctrl_options
-        # create ssh control master command
-        cmd = ["ssh"] + ctrl_options + [self.host, "exit"]
+        if not exists(self.ctrl_path):
```

Comment:
It happens to me from time to time (after resume from sleep?) that I have a socket file but it is stable - attempt to login just hangs until I remove it. But I am not sure if anything could be done besides some check with a timeout